core/src/main/java/com/sequenceiq/cloudbreak/job/stackpatcher/ExistingStackPatcherJobService.java

```diff
@@ -80,21 +91,37 @@ private JobDetail buildJobDetail(ExistingStackPatcherJobAdapter resource) {
                 .build();
     }
 
-    private Trigger buildJobTrigger(JobDetail jobDetail) {
+    private Trigger buildJobTrigger(JobDetail jobDetail, StackPatchType stackPatchType) {
         return TriggerBuilder.newTrigger()
                 .forJob(jobDetail)
                 .withIdentity(jobDetail.getKey().getName(), TRIGGER_GROUP)
-                .withDescription("Trigger for patching existing stack")
-                .startAt(delayedFirstStart())
+                .withDescription("Trigger for existing stack patch " + stackPatchType.name())
                 .withSchedule(SimpleScheduleBuilder.simpleSchedule()
-                        .withIntervalInHours(properties.getIntervalInHours())
+                        .withIntervalInMinutes(getIntervalInMinutes(stackPatchType))
                         .repeatForever()
                         .withMisfireHandlingInstructionNextWithExistingCount())
+                .startAt(getFirstStart(stackPatchType))
                 .build();
     }
 
-    private Date delayedFirstStart() {
-        int delayInMinutes = RANDOM.nextInt((int) TimeUnit.HOURS.toMinutes(properties.getMaxInitialStartDelayInHours()));
-        return Date.from(ZonedDateTime.now().toInstant().plus(Duration.ofMinutes(delayInMinutes)));
+    private int getIntervalInMinutes(StackPatchType stackPatchType) {
```

Comment:
Can't we have this time calculation magic in the patcher implementation so the production code should not be aware of we are mocking or not?